docs/contribution_guidelines.rst

```diff
@@ -54,13 +126,79 @@ All source code files should have license headers. To automatically add the head
 
 .. note::
 
-If you checkin third-party code or files, do not add license headers for them. Follow these instructions to ensure we
-are compliant with Apache Licensing process. <TBD>
+If you checkin third-party code or files, ensure they are distributed under Apache, MIT or BSD-3 licenses.
```

Comment:
Let's put the link for the below page. It gives us the extensive list of allowed/not-allowed licenses
https://www.apache.org/legal/resolved.html#what-can-we-include-in-an-asf-project-category-a
https://www.apache.org/legal/resolved.html#what-can-we-maybe-include-in-an-asf-project-category-b
https://www.apache.org/legal/resolved.html#what-can-we-not-include-in-an-asf-project-category-x

Reply:
Added